Added Speedtest artifact extractor

its5Q opened the pull request:
This PR implements a plugin to extract artifacts from the Speedtest app, which contain a lot of useful data such as location data and network names. Inspired by the recently finished Belkasoft CTF :)

Test data from the CTF image: [org.zwanoo.android.speedtest.zip](https://github.com/user-attachments/files/21455127/org.zwanoo.android.speedtest.zip)

Resulting report:
<img width="2560" height="1331" alt="chrome_cVDfOxhYH6" src="https://github.com/user-attachments/assets/5cfdad34-c192-477c-bb91-b51a2b869528" />
<img width="2560" height="1331" alt="image" src="https://github.com/user-attachments/assets/90db28dc-40e0-47d7-8ece-d871f17cafb0" />

its5Q commented:
Looks good now
<img width="986" height="132" alt="EmEditor_y6Z4Bnu24J" src="https://github.com/user-attachments/assets/6578129c-8b73-41c3-a812-4474aec86dfe" />
<img width="1008" height="539" alt="EmEditor_GmksgmVLvm" src="https://github.com/user-attachments/assets/9daf7ac8-0727-455e-bde7-e362ed03ce07" />

stark4n6 commented:
@its5Q do you want to try doing LAVA compliance reports? You can read about docs here:
[https://github.com/abrignoni/iLEAPP/blob/main/admin/docs/module_updates.md](https://github.com/abrignoni/iLEAPP/blob/main/admin/docs/module_updates.md)
[https://github.com/abrignoni/iLEAPP/blob/main/admin/docs/module_updates_advanced.md](https://github.com/abrignoni/iLEAPP/blob/main/admin/docs/module_updates_advanced.md)

its5Q commented:
Hm, yeah, sure. I just went with the manual report generation because I wanted to merge two tables into one HTML report, but I guess it's fine to split that up and switch to automatic report generation for this module as well. I'll get on that probably tomorrow.

abrignoni commented:
Here is a video on how to make LAVA compliant artifacts:
https://www.youtube.com/live/qTgZUh4GPxk?feature=shared

On Mon, Aug 4, 2025 at 1:36 PM its5Q ***@***.***> wrote:

> *its5Q* left a comment (abrignoni/ALEAPP#582)
> <https://github.com/abrignoni/ALEAPP/pull/582#issuecomment-3151772676>
>
> Hm, yeah, sure. I just went with the manual report generation because I
> wanted to merge two tables into one HTML report, but I guess it's fine to
> split that up and switch to automatic report generation for this module as
> well. I'll get on that probably tomorrow.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/abrignoni/ALEAPP/pull/582#issuecomment-3151772676>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG3DPC7YPR7OCO4USSRN3CT3L6KZLAVCNFSM6AAAAACCPB5BKGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCNJRG43TENRXGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

its5Q commented:
@abrigoni I've switched to artifact_processor and renamed some headers so the KML generation works for location data. Also, I've made a tiny change in the OS version module to strip the newline, before that, a trailing newline would end up in the "Build version" field:
<img width="341" height="129" alt="EmEditor_3Asa3tqllk" src="https://github.com/user-attachments/assets/abb8be53-0dfc-4359-947a-5f6e9662dc73" />

its5Q commented:
Yeah, that works. Also made some changes to output all timestamps in UTC.
